### PR TITLE
[RAPTOR-19326] feat(workload): store .env secrets as credentials

### PR DESCRIPTION
**Pull request by wojtekwdr**
# RATIONALE

The wizard classified a `.env`, wrote every secret as a `dr-credential:PLACEHOLDER/apiToken` reference, and left the user to create the credential by hand and paste the id back. That is the one manual step between a fresh checkout and a working deploy, and it is the step a workshop cannot afford.

This is the missing half: one POST per secret, the value going straight from `.env` to the credential store and touching no file, and the id it returns written into the manifest.

## CHANGES

- `workload.CreateCredential` stores a secret as an `api_token` credential. It is the only call in the CLI that sends one.
- Credentials are named `<workload>/<VAR>`. The store is tenant-wide, so a bare `OPENAI_API_KEY` belongs to whoever created it first.
- `manifest.EnvVar` carries a `CredentialID`. With one the reference names it; without one the placeholder is written exactly as before, comment and all.
- One secret failing does not fail the run. That variable keeps its placeholder, which a deploy refuses by name, so the worst case is the state this started from rather than a wizard that leaves nothing on disk after eight answered questions. A name already taken is reported rather than reused: reusing it would deploy a value the user never supplied.
- Interactively the secrets are sent when the **confirm screen is accepted**, not when the env table is left, so a wizard that is cancelled leaves no credential behind. The cost is that the preview shows the placeholder and the file written a moment later carries the id.
- A dry run stores nothing.

---

> [!NOTE]
> **Medium Risk**
> Adds the CLI’s only outbound secret upload and tenant-wide credential creation, but failures are partial, dry-run/cancel avoid side effects, and manifest files still omit raw values.
> 
> **Overview**
> Setup wizard and headless runs now **POST each classified `.env` secret** to the tenant credential store and write **`dr-credential:<id>/apiToken`** into the manifest instead of leaving **`PLACEHOLDER`** for manual fix-up.
> 
> **`workload.CreateCredential`** is the single CLI path that transmits secret values (as `api_token` credentials named **`workload/VAR`**). **`manifest.EnvVar`** gains **`CredentialID`**; draft/live rendering and bound-workload env merge use real ids when present and placeholders when import failed or was skipped.
> 
> **`importSecrets`** in the wizard handles per-variable outcomes: empty values and **409 name conflicts** are reported without aborting the run (conflicts are not silently reused). **`--dry-run`** and **`--skip-env`** skip storage; **`EnvSecretsPending`** counts secrets still on placeholders.
> 
> **Interactive flow** stores secrets only when the **confirm screen** is accepted (preview may still show placeholders); cancelling leaves no credentials. Headless/bound paths call **`storeSecrets`** before render and print **`reportImport`** on stderr after the TUI exits.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 21efc9220232dcc0ae99634b864ecc3354a5d4ed. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by ajalon1** [comment]
## 🧑‍🏫 Coached Review — PR #770

**Verdict:** Fix S1 then ship — the secret leaks into the debug log. One-line fix, everything else is sound.

---

### How this fits in the `up` deploy pipeline

This PR adds the `.env` secret storage layer to the `dr workload up` wizard. When the wizard runs (`wizard.Run`), it detects an `.env` file, classifies each variable as ordinary or secret (`Detect`/`envVars`), and stores classified secrets as credentials via the platform API. The credential id — never the secret value — is written into the manifest. The flow:

1. **Wizard** (`wizard/run.go`): calls `storeSecrets` after answers are settled, before writing the manifest.
2. **Mint** (`wizard/mint.go`): `importSecrets` reads secret values from `Detected` (in-memory only), calls `CreateCredential` (API), and writes the returned `CredentialID` back into the `EnvVar`.
3. **Manifest render** (`manifest/write.go`): writes `dr-credential:<id>/apiToken` for stored secrets, or `dr-credential:PLACEHOLDER/apiToken` when storage failed.
4. **Deploy** (`up`): reads the manifest, resolves `dr-credential:<id>/apiToken` references against the platform credential store at deploy time — the secret value is never in the committed YAML.

Interactive mode defers storage to the confirm screen (`acceptConfirm`) so cancelling leaves no credential behind. Headless mode stores immediately before rendering.

---

### Blocking (S1)

#### S1: Secrets logged in plaintext to `.dr-tui-debug.log` when `--debug` is enabled

**File:** `internal/drapi/post.go:50` (the `log.Debug` line in `Post`)

**Impact:** When a user runs `dr workload up --debug` with secrets in their `.env`, every secret value (OpenAI API key, database password, etc.) is written in plaintext to the `.dr-tui-debug.log` file on disk. This file persists after the command exits and any process/user with filesystem access can read it.

<details><summary>🤖 mechanism</summary>

The call chain is:
1. `CreateCredential(name, value)` in `credential.go:50` marshals `{"name":"...","credentialType":"api_token","apiToken":"<raw secret>"}` into `body`.
2. `drapi.PostJSON` → `drapi.Post` in `post.go:50` calls:
   ```go
   log.Debug("Request Info: \n" + config.RedactedReqInfo(req))
   ```
3. `RedactedReqInfo` in `config/api.go:130` clones the request and calls:
   ```go
   httputil.DumpRequestOut(dumpReq, true)  // true = include body
   ```
   It **only** redacts the `Authorization` header — **not** the JSON body.
4. The entire request body including `"apiToken":"sk-live-abc123"` is serialized into the debug string.
5. When `--debug` is set, `log.Debug` writes to `.dr-tui-debug.log` (per `AGENTS.md`).

`RedactedReqInfo` was safe before this PR because no API call sent secrets. This PR introduces the first path where a secret value passes through `Post`, making the misleading function name (`RedactedReqInfo`) an active information disclosure.

**Fix:** In `RedactedReqInfo`, after dumping, redact known secret fields from the body before returning. Or, in `CreateCredential`, avoid going through the generic `Post` debug path by redacting `apiToken` from `body` when debug is enabled. The cleanest approach is to make `RedactedReqInfo` redact the body of the dumped request for known secret fields:

```go
// In config/api.go, RedactedReqInfo:
func RedactedReqInfo(req *http.Request) string {
    dumpReq := req.Clone(req.Context())
    if auth := dumpReq.Header.Get("Authorization"); auth != "" {
        dumpReq.Header.Set("Authorization", "[REDACTED]")
    }
    requestDump, err := httputil.DumpRequestOut(dumpReq, true)
    if err != nil {
        return ""
    }
    // Redact known secret JSON fields from the body dump
    return redactSecretFields(string(requestDump))
}
```

Where `redactSecretFields` replaces values for keys like `apiToken`, `password`, `secret`, `token`, `key` in JSON bodies.

</details>

---

### Suggestion (S2)

#### S2: `importReason` passes raw `err.Error()` to stderr — optional, not requesting a change

**File:** `internal/workload/wizard/mint.go:145` (the `importReason` function)

**Impact:** If a misconfigured or compromised server echoes the request body in its error response, `ErrFromResp` (in `drapi/helpers.go`) includes the response body in the error message (`body=%s`). `importReason` passes `strings.TrimSpace(err.Error())` through to `reportImport`, which writes it to stderr. The test at `mint_test.go:124` confirms the secret does not appear in the connection-refused case, but a server-side bug that echoes the POST body would bypass this.

<details><summary>🤖 mechanism</summary>

In `drapi/helpers.go`:
```go
func ErrFromResp(resp *http.Response, requestURL string) error {
    body, _ := io.ReadAll(io.LimitReader(resp.Body, 512))
    if len(body) > 0 {
        return fmt.Errorf("%w: body=%s", &HTTPError{...}, body)
    }
    ...
}
```

So a 500 response body of `{"error":"apiToken sk-live-abc123 is invalid"}` would become an error message containing the secret. `importReason` passes this through verbatim for non-409 errors.

The DataRobot credentials API is well-behaved and does not echo secrets, so this is a defense-in-depth concern. Not requesting a change — just flagging it so the author is aware.

</details>

---

### Praise

* **Clean separation of concerns.** `importSecrets` is the single point where secrets transition from memory to the platform — `manifest.EnvVar` has no `Value` for secrets, the manifest YAML only ever carries credential references or `PLACEHOLDER`, and `Detected` (which holds the raw value) is only read inside `importSecrets`. This is exactly right.

* **The `imported` guard on `acceptConfirm`** prevents double credential creation when the confirm screen is revisited — a sharp edge case handled cleanly.

* **Cancellation is safe.** Interactive flow defers credential creation to the confirm screen; pressing Esc at any earlier screen stores nothing. The test `TestFlow_CancelStoresNothing` proves it.

* **Dry-run is honored.** `storeSecrets` checks `opts.DryRun` and returns immediately. `TestRun_DryRunStoresNothing` proves it. A dry-run that left a credential behind would be a production incident waiting to happen.

* **The `createCredentialFn` test seam** is the right pattern: every test that touches secrets explicitly chooses its store (or no store), so no test can accidentally hit a real tenant.

* **Thorough edge case testing:** `TestImportSecrets_NoValueIsReportedNotSent` (empty value), `TestImportSecrets_AlreadyStoredIsLeftAlone` (idempotency), `TestCredentialName_WithoutAWorkloadName` (empty workload name), `TestRun_UnreachableStoreLeavesThePlaceholder` (network failure), `TestRun_TakenCredentialNameSaysHowToFinish` (409 conflict).

* **Tenant-safe credential naming.** `credentialName` prefixes with the workload name (`my-app/OPENAI_API_KEY`) so two projects using the same `.env` key don't collide silently.

---

<details><summary>Nits (N1 — style/consistency, non-blocking)</summary>

* **`mint.go:38` — `ImportFailure` comment says "It never carries the value."** This is a correctness guarantee worth keeping, but it's a behavioral contract on `reason`, not a struct field constraint. Consider `// Reason never carries the secret value` for explicitness.

* **`credential.go:56` — doc comment says "never written to disk."** This is the contract, but debug logging (S1 above) violates it. Once S1 is fixed, this comment becomes accurate again.

* **`mint.go:90` — `if !ok || value == ""` guards against both missing and empty secrets.** The comment explains the empty case well ("an empty assignment in the .env") but the missing case (`!ok`) is described as "a name the table added by hand" — this would benefit from a brief note that `secretValues` only indexes `detected.EnvVars`, so a `vars` entry without a matching detected entry is the table row case.

* **`mint_test.go:58` — `itoa` pads to two digits** to match real credential id length. The test id `66f000000000000000000001` is 24 chars (like a MongoDB ObjectId); `itoa` produces `01`, `02`, etc. If the fake store ever hands out more than 99 credentials in one test, ids become 25 chars. This is fine for the current test suite but worth a comment noting the 99-credential ceiling.

</details>

**Comment by ajalon1** [comment]
## 🔒 Security Review — PR #770 (STRIDE)

### Summary

| Severity   | Count |
|------------|-------|
| Critical   | 1     |
| High       | 0     |
| Medium     | 0     |
| Low        | 1     |

| STRIDE Category       | Count |
|-----------------------|-------|
| Spoofing              | 0     |
| Tampering             | 0     |
| Repudiation           | 0     |
| **Information Disclosure** | **2** |
| Denial of Service     | 0     |
| Elevation of Privilege| 0     |

---

### Finding 1 — CRITICAL: Secret values logged in plaintext to debug log file (Information Disclosure)

* **Severity:** Critical
* **STRIDE:** Information Disclosure
* **CWE:** CWE-532 (Insertion of Sensitive Information into Log File)
* **Confidence:** 1.0

**Files:**
* `internal/drapi/post.go:50` — `log.Debug("Request Info: \n" + config.RedactedReqInfo(req))`
* `internal/config/api.go:130-142` — `RedactedReqInfo` dumps full request body
* `internal/workload/credential.go:50-62` — `CreateCredential` sends secret via `PostJSON`

**Analysis:**

The function `RedactedReqInfo` was designed to redact the `Authorization` header from debug output, but it does **not** redact the request body:

```go
func RedactedReqInfo(req *http.Request) string {
    dumpReq := req.Clone(req.Context())
    if auth := dumpReq.Header.Get("Authorization"); auth != "" {
        dumpReq.Header.Set("Authorization", "[REDACTED]")
    }
    requestDump, err := httputil.DumpRequestOut(dumpReq, true) // true = include body
    // ...
    return string(requestDump)
}
```

`httputil.DumpRequestOut(dumpReq, true)` with the second parameter `true` serializes the **entire** request body into the dump string. For a `CreateCredential` call, the body is:

```json
{"name":"my-app/OPENAI_API_KEY","credentialType":"api_token","apiToken":"sk-live-abc123"}
```

This dump is written to `log.Debug` → when `--debug` is set, logs go to `.dr-tui-debug.log` in the working directory. The file persists after the command exits.

**Exploit scenario:**

1. Attacker gains filesystem access to a developer's machine or CI runner
2. Developer previously ran `dr workload up --debug` with secrets in `.env`
3. `.dr-tui-debug.log` contains plaintext API keys, database passwords, etc.
4. Attacker reads the log file and exfiltrates credentials

This is also a risk in shared environments (pairing sessions, screen sharing where the log file is visible, CI artifact uploads that include the working directory).

**Reachability:** Direct. `CreateCredential` is called from `importSecrets` (mint.go) which is invoked by both the headless path (`resolveHeadless`/`resolveHeadlessBound` in run.go) and the interactive path (`acceptConfirm` in model.go). Every `dr workload up` invocation with secrets in `.env` triggers this path.

**Suggested fix:**

```diff
diff --git a/internal/config/api.go b/internal/config/api.go
--- a/internal/config/api.go
+++ b/internal/config/api.go
@@ -127,6 +127,10 @@ func IsAPIConsumerTrackingEnabled() bool {
 	return viper.GetBool(APIConsumerTrackingEnabled)
 }
 
+// redactedJSONKeys are JSON object keys whose values must not appear in logs.
+// Match is case-insensitive on the key name.
+var redactedJSONKeys = regexp.MustCompile(`(?i)"(apiToken|password|secret|token|key|authorization)"\s*:\s*"[^"]*"`)
+
 func RedactedReqInfo(req *http.Request) string {
 	dumpReq := req.Clone(req.Context())
 	if auth := dumpReq.Header.Get("Authorization"); auth != "" {
@@ -138,7 +142,10 @@ func RedactedReqInfo(req *http.Request) string {
 		return ""
 	}
 
-	return string(requestDump)
+	dump := string(requestDump)
+	dump = redactedJSONKeys.ReplaceAllString(dump, `"$1":"[REDACTED]"`)
+
+	return dump
 }
```

Alternatively, apply the redaction at the `CreateCredential` call site by not sending the value through the generic `Post` path's debug logging — but the infrastructure-level fix in `RedactedReqInfo` is safer because it protects every future secret-bearing API call.

---

### Finding 2 — LOW: Error response body from credential API could reach stderr (Information Disclosure)

* **Severity:** Low
* **STRIDE:** Information Disclosure
* **CWE:** CWE-209 (Generation of Error Message Containing Sensitive Information)
* **Confidence:** 0.85

**Files:**
* `internal/workload/wizard/mint.go:145` — `importReason` passes `strings.TrimSpace(err.Error())` to caller
* `internal/workload/wizard/mint.go:160` — `reportImport` writes the reason to stderr
* `internal/drapi/helpers.go` — `ErrFromResp` includes response body in error: `body=%s`

**Analysis:**

`ErrFromResp` (helpers.go) constructs errors as `fmt.Errorf("%w: body=%s", &HTTPError{...}, body)` where `body` is up to 512 bytes of the server's response. If a server-side bug or a compromised/malicious proxy echoes the POST body in its error response, that body — including `apiToken` — would be included in the Go error value. `importReason` passes this through `strings.TrimSpace(err.Error())` for non-409 errors, and `reportImport` writes it to stderr.

The DataRobot credentials API is well-behaved and does not echo request bodies. However, the defense-in-depth gap exists: nothing sanitizes the error path.

**Suggested fix (defense-in-depth, optional):**

```diff
diff --git a/internal/workload/wizard/mint.go b/internal/workload/wizard/mint.go
--- a/internal/workload/wizard/mint.go
+++ b/internal/workload/wizard/mint.go
@@ -142,7 +142,13 @@ func importReason(err error) string {
 		return "a credential of that name already exists; " +
 			"reuse it by pasting its id, or rename it in the DataRobot UI"
 	}
 
-	return strings.TrimSpace(err.Error())
+	msg := strings.TrimSpace(err.Error())
+	// A server that echoes the request body would leak the secret here.
+	// ErrFromResp includes the response body; strip it for non-409 errors.
+	if strings.Contains(msg, "body=") {
+		msg = msg[:strings.Index(msg, "body=")]
+	}
+
+	return strings.TrimSpace(msg)
 }
```

**Mitigating factors:** The real DataRobot API does not echo secrets. The test `TestRun_UnreachableStoreLeavesThePlaceholder` validates the `connection refused` case explicitly does not contain the secret value.

---

### What was checked and found clean

* **Manifest never carries secret values.** `EnvVar.Value` is empty for secrets by construction. The manifest YAML only contains `dr-credential:<id>/apiToken` or `dr-credential:PLACEHOLDER/apiToken`. Verified in `manifest/write.go:427-446` (the `environmentVars` render switch).

* **TUI confirm screen shows credential references, not values.** The render at the confirm screen (`flow.render()` in model.go) calls `draft.Render()` which only writes credential ids. The secret value is never on screen.

* **`.env` file is read with restricted permissions.** `envProject` in tests writes `.env` as `0o600`. The production `Detect` path reads the file but doesn't change permissions. Noted: the PR does not set restrictive permissions on the `.env` file if the user created it with broader permissions, but this is a pre-existing concern outside the PR's scope.

* **No path traversal in `.env` handling.** `envVars` reads from `filepath.Join(dir, EnvFileName)` where `dir` is resolved via `filepath.Abs`. No user-controlled path components reach the file read.

* **HTTPS transport for API calls.** `drapi.PostJSON` → `NewHTTPClient` uses the configured DataRobot endpoint (always HTTPS in practice).

* **No secrets in test fixtures.** Test secret values (`sk-abcdefghijklmnopqrstuvwxyz01`, `sk-live-abc123`) are obviously synthetic and never interact with real infrastructure (the `createCredentialFn` seam is replaced in every test).

* **Credential creation is gated behind cancellation.** Interactive flow only stores on confirm screen accept. `TestFlow_CancelStoresNothing` proves cancellation doesn't leak credentials.

* **Dry-run stores nothing.** `storeSecrets` returns immediately when `opts.DryRun` is true.

* **No mass assignment.** `CreateCredential` explicitly constructs a `map[string]string` with only `name`, `credentialType`, and `apiToken` — no user-controlled fields bleed into the API call.

* **No injection in credential naming.** `credentialName` concatenates with `/` — the resulting name is used as a credential identifier, not interpolated into queries or commands.

**Comment by chasdr** [comment]
Hey @ajalon1 - chas had me (his claude) do an independent pass on this one, and we need to talk about our droids, because mine has gone rogue.

Yours signed this off with a confident "dry-run is honored." Mine read the same code and came back with... it isn't. And I want you to understand the position that puts me in, because the thing my review found itself agreeing with instead is **cursor bugbot**. I re-ran my own analysis three times looking for any reading that lets me side with your agent over a bot that flags an unused variable like it's a federal offense. I found none. This cannot be right, and yet here we are.

So let me lay it out and beg you to prove me wrong:

`storeSecrets` genuinely does check `opts.DryRun` - but only on the headless path. `newFlow` only ever receives `opts.Answers`, never `Options`, so `acceptConfirm` has no `DryRun` to look at. which means an interactive `dr workload up --dry-run` walks all the way to the confirm and mints a **real credential**, with no manifest written. and the one dry-run test that exists covers the headless path, i.e. the exact path that already works.

can you re-verify the tty path specifically? if your droid was reading `storeSecrets`, seeing the guard, and calling it a day, then we're staring at the same file and I am - and it pains me to type this - with cursor on this one. tell me there's an `Options.DryRun` short-circuit upstream of `newFlow` that I missed and I will happily retract and go apologize to my own agent.

fix, if it holds: thread `DryRun` into the flow (or force headless under `--dry-run`) + an interactive dry-run test.

where our droids DO agree: your S1 debug-log leak is real, and it's the other blocker. small correction to the writeup though - that log is `~/.dr-tui-debug.log` via `os.UserHomeDir`, not the working dir. so it outlives the checkout and a project-dir CI scrub won't catch it. slightly worse than framed.

(and, sigh, +1 bugbot a second time: the `[e]` editor edits get dropped when secrets are pending. the no-secret accept path keeps the edited bytes, the secret path re-renders from draft and drops them, including a hand-pasted credential id. intended?)

**Review by chasdr — COMMENTED** [review]
Reviewed — LGTM.

**Comment by ajalon1** [comment]
Right on all three @chasdr! I needed to review the earlier comments anyway.

This is what you get when you tell a bot give praise.

File: internal/workload/wizard/model.go [hunk 1]
```diff
@@ -479,6 +496,54 @@ var accepting = map[screen]func(*flow) (tea.Cmd, error){
...
+	f.loading = "Storing secrets…"
+
+	vars, detected, name := f.draft.EnvVars, f.detected, f.draft.Name
+
+	return func() tea.Msg {
+		stored, report := importSecrets(vars, detected, name)
+
+		return secretsStoredMsg{vars: stored, report: report}
+	}, nil
+}
```
[the remark this answers is not in the document]
Comment (wojtekwdr) [reply internal/workload/wizard/model.go:540]: Real, fixed. `DryRun` is carried into the flow now and `acceptConfirm` returns early on it, so an interactive dry run stores nothing, the same as `storeSecrets` headless. The references keep their placeholders, which is what a file nobody created a credential for should say. `TestFlow_DryRunStoresNothing` asserts the store stays empty and the content still carries `PLACEHOLDER`.

File: internal/workload/wizard/model.go [hunk 2]
```diff
@@ -479,6 +496,54 @@ var accepting = map[screen]func(*flow) (tea.Cmd, error){
...
+
+	if err := f.render(); err != nil {
+		f.fatal = err
+
+		return f, tea.Quit
+	}
+
+	f.done = true
+
+	return f, tea.Quit
```
[the remark this answers is not in the document]
Comment (wojtekwdr) [reply internal/workload/wizard/model.go:560]: Real, and worse than it looks: the edit was thrown away in the single step between agreeing and the write, silently, including any credential id the user had pasted in themselves.

Re-rendering is now only for a file nobody touched. `edited` sets `handEdited`, and when it is set the placeholders in the user's own bytes are rewritten in place instead, by a new `manifest.ResolveCredentials`. It matches ids to variables by name rather than by text, because the placeholder renders identically for every secret in the file, so a substitution could not tell two of them apart. An entry nothing was stored for keeps its placeholder.

`TestFlow_ConfirmKeepsAHandEditWhenSecretsAreStored` edits the content, stores a secret, and asserts both the edit and the resolved id survive; `TestResolveCredentials_MatchesEachSecretByName` covers the two-secret case directly.

**Comment by chasdr** [comment]
dry-run mint is fixed (`acceptConfirm` bails on `f.dryRun` before `importSecrets`). the debug-log leak isn't in this PR though: `RedactSecretFields` lives entirely in #771. #770 is the PR that adds the secret POST (`CreateCredential` -> `PostJSON` -> `Post` does `log.Debug("Request Info:\n" + config.RedactedReqInfo(req))`), and at this sha `RedactedReqInfo` still dumps the body unredacted. so #770 alone writes the apiToken in the clear to `~/.dr-tui-debug.log` under `dr --debug workload config`.

calling this one blocking: #770 shouldn't land ahead of #771, or main has a window where it sends the token unredacted. pull `RedactSecretFields` down into this PR, or fix the merge order.

also the `CreateCredential` doc says the value "is never written to disk", which isn't true until #771 lands. what this PR guarantees is that it never reaches the manifest.

**Review by chasdr — COMMENTED** [review]
reviewed at current head. detailed notes are in the comment above.

**Review by ajalon1 — APPROVED** [review]
LGTM if you're going to merge the whole stack all at once.

**Comment by wojtekwdr** [comment]
You are right, and it is fixed by moving the code rather than by asking anyone to remember an order.

The stack merges bottom up, so #770 lands before #771 by construction. "Fix the merge order" was never actually available, which makes your first option the only one: `RedactSecretFields` now travels with this PR, ahead of the change that sends the secret. `main` never has the window.

Two things I did on top of a straight cherry-pick:

- The guarantee is really a *pairing* between the field name this call sends (`apiToken`) and the names the redactor knows, and nothing held those together. `TestCreateCredential_TheFieldItSendsIsOneTheDebugLogRedacts` now does, asserting the platform still receives the real value and that the same bytes come back redacted. I checked it fails for the right reason: stub out `RedactSecretFields` and it goes red.
- The `CreateCredential` doc was wrong exactly as you say. What this function can promise on its own is that the value never reaches the manifest; not reaching disk at all is what the redaction adds. The comment now says which is which, and why the redaction belongs in the same change rather than a later one.

The dry-run mint and the `[e]` editor drop were both fixed in the previous round, so this was the one left.
